openlibrary/plugins/importapi/code.py

```diff
@@ -338,24 +339,33 @@ def get_ia_record(metadata: dict) -> dict:
         lccn = metadata.get('lccn')
         subject = metadata.get('subject')
         oclc = metadata.get('oclc-id')
+        number_of_pages = metadata.get('imagecount')
         d = {
-            'title': metadata.get('title', ''),
+            'title': metadata.get('title', '').strip(),
             'authors': authors,
-            'publish_date': metadata.get('date'),
-            'publisher': metadata.get('publisher'),
+            'publish_date': metadata.get('date', '').strip(),
+            'publisher': metadata.get('publisher', '').strip(),
```

Comment:
Do we need this `strip`s? This changes the behaviour a bit, since previously `publish_date` could be `None`. I'm not sure how great the tests are here so rolling back might be easier! Have you noticed this being a problem?

```suggestion
            'publish_date': metadata.get('date'),
            'publisher': metadata.get('publisher'),
```

Reply:
Good point that I was recklessly changing the behaviour. Thank you for catching that. :)

I only noticed one record in the test set that had an issue ([The First Year of Life - Special Early Learning Supplement](https://openlibrary.org/books/OL46523894M.json)), but I did a quick scan of the latest dump and there appear to be 142,386 editions where there is a space before or after a title.

That said, [they](https://openlibrary.org/books/OL12565688M.json) [all](https://openlibrary.org/books/OL8986768M.json) [seem](https://openlibrary.org/books/OL11146051M.json) [to](https://openlibrary.org/books/OL12655866M.json) be created around 2008-04-30, so something could have gone wrong that day. Though I just spot checked a few and all within about 10,000 or so of each other.

I will just roll this one back.